LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -6,172 +6,239 @@ namespace LearningMissionSimulation
             {
                 Account account = ObjectGenerator.GenerateAccount();
-                ReportItem(account.ToString(), "Created account", i++);
-                accountDictionary.Add(account.Id, account);
+                _accountDictionary.Add(account.Id, account);
+
+                ReportItem(itemName: account.ToString(), actionName: action, itemIndex: i);
+
                 if (account.Role == Role.Student)
                 {
                     Student student = ObjectGenerator.GenerateStudent(account.Id);
-                    studentList.Add(student);
+                    _studentDictionary.Add(account.Id, student);
                     if (account.Status == Status.Pending)
                     {
-                        pendingAccountQueue.Enqueue(account);
+                        _pendingAccountQueue.Enqueue(account);
+                    }
+                    else if(account.Status == Status.Active)
+                    {
+                        _activeStudentList.Add(student);
                     }
                 }
                 else if(account.Role == Role.Instructor)
                 {
                     Instructor instructor = ObjectGenerator.GenerateInstructor(account.Id);
-                    instructorList.Add(instructor);
+                    _instructorDictionary.Add(account.Id, instructor);
                     if(account.Status == Status.Pending)
                     {
-                        pendingAccountQueue.Enqueue(account);
+                        _pendingAccountQueue.Enqueue(account);
+                    }
+                    else if (account.Status == Status.Active)
+                    {
+                        _activeInstructorList.Add(instructor);
                     }
                 }
                 i++;
             }
-            ReportSummary("Generated", "accounts", accountCount);
-            ReportFooter("Create Account");
+
+            ReportSummary(actionName: action, itemCount: accountCount);
+            ReportFooter(actionName: action);
         }
 
         public void ActivateAccounts()
         {
-            ReportHeader("Activate account");
-            int pendingAccountQueueCount = pendingAccountQueue.Count;
+            string action = "Activate account";
+            ReportHeader(actionName: action);
+
+            int pendingAccountQueueCount = _pendingAccountQueue.Count;
             int i = 0;
-            while(pendingAccountQueue.Count > 0)
+            while(_pendingAccountQueue.Count > 0)
             {
-                Account account = pendingAccountQueue.Dequeue();
+                Account account = _pendingAccountQueue.Dequeue();
                 account.Status = Status.Active;
-                ReportItem(account.ToString(), "Activated account", i++);
+                if(account.Role == Role.Student)
+                {
+                    Student student;
+                    _studentDictionary.TryGetValue(account.Id, out student);
```

Comment:
Do not assume that _studentDictionary contains the student object, check for null, add student to _activeStudentList only if student!=null. Do the same for instructors.